### PR TITLE
AMBR-1118 Use env vars for configuration

### PR DESCRIPTION
**Pull request by egh**
https://jira.plos.org/jira/browse/AMBR-1118

Modifies rhino to use environment variables for configuration. 
- Remove old debian packaging 348a68f
- Remove unused configuration
- Hardcode some values that we previously configurable at runtime (34b8a61 eac2452 b4a8658)
- Remove kafka (98f4f09)
- Upgrade to tomcat 5.5.50 for local dev
- Remove ability to specify which bucket to use (5af4db3) since all corpus content is stored in the same bucket.

See the updated README for how to run:
https://github.com/PLOS/rhino/pull/512/files#diff-04c6e90faac2675aa89e2176d2eec7d8R16

### CODE REVIEW

**Comment by sgatchell** [comment]
cleanup bad! :stuck_out_tongue: 
image

**Comment by egh** [comment]
FYI @sgatchell I decided to ensure that logging was configurable as well. In the process of doing that I removed sl4j, which seems to not be compatible with log4j2, which we use.


0fbfef3 

You can now tweak the log levels using the `LOG_LEVEL` env var. Try `LOG_LEVEL=debug`

**Comment by sgatchell** [comment]
image :feelsgood: this is exciting

**Comment by egh** [comment]
> Thanks for killing this. I did sometimes wonder where it was used.

I think it was going to be used for preprints or something? @mcbrandt13 might now. But I never understood why we should store preprints in a separate bucket. In any case, I'm glad you're happy it's gone. I was worried that I was doing too much extra work eliminating it.

File: src/main/java/org/ambraproject/rhino/service/taxonomy/impl/TaxonomyClassificationServiceImpl.java [hunk 1]
```diff
@@ -122,12 +120,13 @@
   @Autowired
   protected HibernateTemplate hibernateTemplate;
 
+  private static final String[] CATEGORY_BLACKLIST =
+      new String[] {"/Earth sciences/Geography/Locations/"};
```
Comment (egh) [inline src/main/java/org/ambraproject/rhino/service/taxonomy/impl/TaxonomyClassificationServiceImpl.java]: See https://jira.plos.org/jira/browse/AMEC-100. You can also look in plos/graveyard under ambra to see the ancient history of this feature. Basically it was a one-time hack which may or may not still be needed. But at some point the hack of blacklisting a single category turned into a configurable feature. It seemed to me that this was absolutely not something that needed to be configurable at runtime, so I just moved it here. It's not clear if this is still a feature we need, but after talking with Jack he felt it was best to keep the feature in for now.

Comment (sgatchell) [reply src/main/java/org/ambraproject/rhino/service/taxonomy/impl/TaxonomyClassificationServiceImpl.java]: Interesting, thanks!  Would it be worth noting the ticket in a comment?

Comment (egh) [reply src/main/java/org/ambraproject/rhino/service/taxonomy/impl/TaxonomyClassificationServiceImpl.java]: Good point, will do.

Comment (egh) [reply src/main/java/org/ambraproject/rhino/service/taxonomy/impl/TaxonomyClassificationServiceImpl.java]: 4f6e354c

File: src/main/java/org/ambraproject/rhino/rest/controller/ConfigurationReadController.java [hunk 2]
```diff
@@ -75,19 +71,4 @@
...
-   * This method exposes the NED login credentials, which is a high security risk.
-   *
-   * @return The NED user/password
-   *
-   * @throws IOException if errors on creating the JSON object
-   *
-   * @deprecated Should not be used, as this is a high security risk
-   */
-  public ResponseEntity<?> readUserApiConfig() throws IOException {
-    // TODO: Remove `readUserApiConfig` method.
```
Comment (sgatchell) [inline src/main/java/org/ambraproject/rhino/rest/controller/ConfigurationReadController.java]: about time, eh?

**Comment by egh** [comment]
@sgatchell Thanks for the review!

**Pull request merged**